packages/dev/core/src/ShadersWGSL/ShadersInclude/lightUboDeclaration.fx

```diff
@@ -0,0 +1,98 @@
+#ifdef LIGHT{X}
+	struct Light{X}
+	{
+		vLightData: vec4f,
+		vLightDiffuse: vec4f,
+		vLightSpecular: vec4f,
+		#ifdef SPOTLIGHT{X}
+			vLightDirection: vec4f,
+			vLightFalloff: vec4f,
+		#elif defined(POINTLIGHT{X})
+			vLightFalloff: vec4f,
+		#elif defined(HEMILIGHT{X})
+			vLightGround: vec3f,
+		#endif
+		shadowsInfo: vec4f,
+		depthValues: vec2f
+	} ;
+
+var<uniform> light{X} : Light{X};
+
+
+#ifdef PROJECTEDLIGHTTEXTURE{X}
+	uniform textureProjectionMatrix{X}: mat4x4f;
+	var projectionLightSampler{X}: sampler;
+#endif
+#ifdef SHADOW{X}
+	#ifdef SHADOWCSM{X}
+		uniform lightMatrix{X}:  array<mat4x4f, SHADOWCSMNUM_CASCADES{X}>;
+		uniform viewFrustumZ{X}:  array<f32, SHADOWCSMNUM_CASCADES{X}>;
+        uniform frustumLengths{X}:  array<f32, SHADOWCSMNUM_CASCADES{X}>;
+        uniform cascadeBlendFactor{X}: f32;
+
+		// Because WGSL does not allow us to have arrays as varying...
+		varying vPositionFromLight{X}_0: vec4f;
+		varying vDepthMetric{X}_0:  f32;
+		varying vPositionFromLight{X}_1: vec4f;
+		varying vDepthMetric{X}_1:  f32;
+		varying vPositionFromLight{X}_2: vec4f;
+		varying vDepthMetric{X}_2:  f32;
+		varying vPositionFromLight{X}_3: vec4f;
+		varying vDepthMetric{X}_3:  f32;
+		varying vPositionFromCamera{X}: vec4f;
+
+		var<private> vPositionFromLight{X}: array<vec4f, 4>;
+		var<private> vDepthMetric{X} : array<f32, 4>;
+
+		#if defined(SHADOWPCSS{X})
+			var shadow{X}Sampler: sampler_comparison;			
+			var shadow{X}Texture: texture_depth_2d_array;
+			var depth{X}Sampler: sampler;
+			var depth{X}Texture: texture_2d_array<f32>;
+            uniform lightSizeUVCorrection{X}: array<vec2f, SHADOWCSMNUM_CASCADES{X}>;
+            uniform depthCorrection{X}: array<f32, SHADOWCSMNUM_CASCADES{X}>;
+            uniform penumbraDarkness{X}: f32;
+		#elif defined(SHADOWPCF{X})
+			var shadow{X}Sampler: sampler_comparison;
+			var shadow{X}Texture: texture_depth_2d_array;
+		#else			
+			var shadow{X}Sampler: sampler;			
+			var shadow{X}Texture: texture_2d_array<f32>;
+		#endif
+
+        #ifdef SHADOWCSMDEBUG{X}
+            const vCascadeColorsMultiplier{X}: array<vec3f, 8> = array<vec3f, 8>
+            (
+                vec3f ( 1.5, 0.0, 0.0 ),
+                vec3f ( 0.0, 1.5, 0.0 ),
+                vec3f ( 0.0, 0.0, 5.5 ),
+                vec3f ( 1.5, 0.0, 5.5 ),
+                vec3f ( 1.5, 1.5, 0.0 ),
+                vec3f ( 1.0, 1.0, 1.0 ),
+                vec3f ( 0.0, 1.0, 5.5 ),
+                vec3f ( 0.5, 3.5, 0.75 )
+            );
+        #endif
+	#elif defined(SHADOWCUBE{X})
+		var shadow{X}Sampler: sampler;		

```

Comment:
Same comment than above.
Also, I think you miss to declare the cube texture?

Reply:
Yes I did!